static/library/slider_component.ts

```diff
@@ -222,74 +315,135 @@ export class DatacommonsSliderComponent extends LitElement {
         </div>
       `;
     }
+    if (this._isLoading || !this._dates) {
+      return html`<div class="container" part="container">Loading...</div>`;
+    }
+
+    const startDate = this._dates[0];
+    const endDate = this._dates[this._dates.length - 1];
+    // Normalized slider value as a percent between 0 and 100
+    const normalizedSliderValue =
+      this._showTrendsSummaryEnabled || this._dates.length <= 1
+        ? 100
+        : (this._value / (this._dates.length - 1)) * 100;
 
+    const dateText = this.getDateText();
+    const endDateText = this.getEndDateText();
+    const lastDateIndex = this._dates.length - 1;
+    const isHighestCoverageDate =
+      dateText === this._highestCoverageDate && !this._showTrendsSummaryEnabled;
+
+    // Text to show under range slider button.
+    // Shows asterisk if showing date of highest coverage
+    let sliderLabelText = this._showTrendsSummaryEnabled ? "" : dateText;
+    if (isHighestCoverageDate) {
+      sliderLabelText += "*";
+    }
     return html`
       <div class="container" part="container">
         ${this.header
           ? html`<h4 part="header">${this.header}</h4>`
           : this.defaultHeader()}
-        <div class="row">
-          <div class="label">${this._dates[0]}</div>
-          <input
-            class="slider"
-            max="${this._dates.length}"
-            min="0"
-            title="${this.getValueText()}"
-            type="range"
-            value="${this._value}"
-            @change=${this.onSliderChange}
-            @input=${this.onSliderInput}
-          />
-          <div class="label">${MOST_RECENT_TEXT}</div>
+
+        <div
+          class="row slider ${this._showTrendsSummaryEnabled ? "disabled" : ""}"
+        >
+          <div class="label">${startDate}</div>
+          <div class="slider-control">
+            <input
+              class="slider"
+              max="${lastDateIndex}"
+              min="0"
+              title="${dateText}"
+              type="range"
+              .value="${this._showTrendsSummaryEnabled
+                ? lastDateIndex
+                : this._value}"
+              @change=${this.onSliderChange}
+              @input=${this.onSliderInput}
+              ?disabled=${this._showTrendsSummaryEnabled}
+            />
+            <div class="slider-label">
+              <div
+                class="slider-label-text"
+                style="left:${normalizedSliderValue}%;"
+              >
+                <span class="slider-label-text-inner">${sliderLabelText}</span>
+              </div>
+            </div>
+          </div>
+          <div class="label">${endDate}</div>
+        </div>
+
+        <div
+          class="row options${!this.showTrendsSummary ? " single-option" : ""}"
+        >
+          ${this.showTrendsSummary
+            ? html` <label class="checkbox-label"
+                ><input
+                  type="checkbox"
+                  @change=${this.onShowLatestChange}
```

Comment:
nit: be consistent about calling this `trends summary` or `latest` even in the code

Reply:
Updated!